qt/python/mantidqt/widgets/fitpropertybrowser/interactive_tool.py

```diff
@@ -0,0 +1,267 @@
+                return i
+        return n + 1
+
+    def add_default_peak(self):
+        self.mouse_state.transition_to('add_peak')
+
+    def add_peak_dialog(self):
+        selected_name = QInputDialog.getItem(self.canvas, 'Fit', 'Select peak function', self.peak_names,
+                                             self.peak_names.index(self.current_peak_type), False)
+        if selected_name[1]:
+            self.peak_type_changed.emit(selected_name[0])
+            self.mouse_state.transition_to('add_peak')
+
+    def add_background_dialog(self):
+        current_index = self.background_names.index(self.default_background)
+        if current_index < 0:
+            current_index = 0
+        selected_name = QInputDialog.getItem(self.canvas, 'Fit', 'Select background function', self.background_names,
+                                             current_index, False)
+        if selected_name[1]:
+            self.add_background_requested.emit(selected_name[0])
+
+    def add_other_dialog(self):
+        selected_name = QInputDialog.getItem(self.canvas, 'Fit', 'Select function', self.other_names, 0, False)
+        if selected_name[1]:
+            self.add_other_requested.emit(selected_name[0])
+
+    def add_peak_marker(self, x, y_top, y_bottom=0.0, fwhm=None):
+        if fwhm is None:
+            fwhm = self.fwhm
+        peak_id = self._make_peak_id()
+        peak = PeakMarker(self.canvas, peak_id, x, y_top, y_bottom, fwhm=fwhm)
+        peak.peak_moved.connect(self.peak_moved)
+        peak.fwhm_changed.connect(self.peak_fwhm_changed_slot)
+        self.peak_markers.append(peak)
+        return peak
+
+    def add_peak(self, x, y_top, y_bottom=0.0):
+        peak = self.add_peak_marker(x, y_top, y_bottom)
+        self.select_peak(peak)
+        self.canvas.draw()
+        self.peak_added.emit(peak.peak_id, x, peak.height(), peak.fwhm())
+
+    def update_peak(self, peak_id, centre, height, fwhm):
+        for pm in self.peak_markers:
+            if pm.peak_id == peak_id:
+                pm.update_peak(centre, height, fwhm)
+        self.canvas.draw()
+
+    def select_peak(self, peak):
+        self.selected_peak = None
+        for pm in self.peak_markers:
+            if peak == pm:
+                pm.select()
+                self.selected_peak = peak
+            else:
+                pm.deselect()
+
+    def _get_default_height(self):
+        ylim = self.ax.get_ylim()
+        return (ylim[0] + ylim[1]) / 2
+
+    def get_peak_list(self):
+        plist = []
+        for pm in self.peak_markers:
+            plist.append((pm.peak_id, pm.centre(), pm.height(), pm.fwhm()))
+        return plist
+
+    def update_peak_markers(self, peaks_to_keep, peaks_to_add):
+        peaks_to_remove = []
+        for i, pm in enumerate(self.peak_markers):
+            if pm.peak_id not in peaks_to_keep:
+                peaks_to_remove.append(i)
+        for i in peaks_to_remove:
+            self.peak_markers[i].remove()
```

Comment:
I was able to cause a "crash" twice with the following back trace:
```
Load-[Notice] Load started
Load-[Notice] Load successful, Duration 0.36 seconds
Fit-[Notice] Fit started
Fit-[Notice] Fit status: success
Fit-[Notice] Stopped after 6 iterations
Fit-[Notice] Fit successful, Duration 0.01 seconds
Fit-[Notice] Fit started
Fit-[Notice] Fit status: success
Fit-[Notice] Stopped after 64 iterations
Fit-[Notice] Fit successful, Duration 1.87 seconds
Traceback (most recent call last):
  File "/media/data1/source/github/mantidproject/mantid/qt/python/mantidqt/widgets/fitpropertybrowser/fitpropertybrowser.py", line 257, in function_changed_slot
    peak_ids, peak_updates = self.tool.update_peak_markers(self.peak_ids.keys(), peaks_to_add)
  File "/media/data1/source/github/mantidproject/mantid/qt/python/mantidqt/widgets/fitpropertybrowser/interactive_tool.py", line 229, in update_peak_markers
    self.peak_markers[i].remove()
IndexError: list index out of range
```
but I can't seem to find the steps I took. I had a Gaussian and an IkedaCarpenterPV function on the same plot and ran a couple of fits and tried to remove one of the functions.